File: lab_python_oop/Rectangle.py
from lab_python_oop.GeometricFigure import GeometricFigure
from lab_python_oop.Color import Color

class Rectangle(GeometricFigure):
    name = "Прямоугольник"

    def __init__(self, width, height, color):
        self.width = width
        self.height = height
        self.color = Color(color)

    def area(self):
        return self.width * self.height

    def __repr__(self):
        return ("{name}: Сторона={side}, Цвет={color}, ""Площадь={area:.2f}").format(
                    name=self.name,
                    side=self.width,
                    color=str(self.color),
                    area=self.area()
                )

File: geometry_project/src/oop/rectangle.rs
use super::{GeometricShape, Color};

pub struct Rectangle {
    width: f64,
    height: f64,
    color: Color,
}

impl Rectangle {
    pub fn new(width: f64, height: f64, color: Color) -> Self {
        Self { width, height, color }
    }
}

impl GeometricShape for Rectangle {
    fn area(&self) -> f64 {
        self.width * self.height
    }

    fn shape_name(&self) -> &'static str {
        "Прямоугольник"
    }
}
